### PR TITLE
encrypt data sent from/to ML Worker

### PR DESCRIPTION
**Pull request by andreybavt**
External worker - ML Worker process diagram
https://miro.com/app/board/uXjVPd1XXws=/?moveToWidget=3458764543811138917&cot=14

- [x] moved from TCP sockets to UNIX socket and Netty's LocalAddress (IPS mechanism) to make it less accessible from other processes on local machines
- [x] Added AES GCM encryption with encryption key passed through a regular http**s** API call. 
- [x] Reading ML Worker port from remote server, not specifying it manually

### CODE REVIEW

**Comment by andreybavt** [comment]
@Googleton , If you have time to check out this PR it'd be great, it can be also a chance for me to share knowledge about the Backend <--> ML Worker data flow

**Comment by Googleton** [comment]
I've actually spent yesterday afternoon on it, we can discuss it whenever
you're available

Le mer. 25 janv. 2023, 23:59, Andrey Avtomonov ***@***.***> a
écrit :

**Comment by Googleton** [comment]
Internal ML Worker doesn't work currently, looks like some flag checks are missing

File: backend/src/main/java/ai/giskard/web/rest/controllers/SettingsController.java [hunk 1]
```diff
@@ -106,10 +110,22 @@ public AppConfigDTO getApplicationSettings(@AuthenticationPrincipal final UserDe
             .build();
     }
 
+    @GetMapping("/ml-worker-connect")
+    public MLWorkerConnectionInfoDTO getMLWorkerConnectionInfo() throws NoSuchAlgorithmException {
```
Comment (Googleton) [inline backend/src/main/java/ai/giskard/web/rest/controllers/SettingsController.java:114]: The exception is never thrown

File: backend/src/main/java/ai/giskard/ml/tunnel/OuterChannelHandler.java [hunk 2]
```diff
@@ -1,175 +1,179 @@
...
+        log.debug("Outer: Writing {} bytes from {}", in.readableBytes(), outerChannel.id());
+        if (channelRegistry.isDataChannel(outerChannel.id())) {
+            passDataDirectly(in, outerChannel);
+        } else {
+            handleServiceChannelInput(ctx, outerChannel, in);
         }
-
     }
 
     private void handleServiceChannelInput(ChannelHandlerContext ctx, SocketChannel outerChannel, ByteBuf in) {
```
Comment (Googleton) [inline backend/src/main/java/ai/giskard/ml/tunnel/OuterChannelHandler.java:90]: ctx is now unused. i think Sonar will catch other things like that but it hasn't ran on the PR yet

**Pull request merged**